sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -148,6 +148,7 @@ class DataSourceWithHiveMetastoreCatalogSuite
 
   override def beforeAll(): Unit = {
     super.beforeAll()
+    sparkSession.sessionState.catalog.reset()
```

Comment:
Hey Xiao, I was curious as to why this would reduce flakiness given that https://github.com/apache/spark/pull/20196 didn't help. Assuming something is changing the current database before/during test execution, how about we just use qualified database + table names here instead (i.e., `saveAsTable("default.t")` and `CREATE TABLE default.t USING ...`)?

Reply:
This is for changing our own current database. The previous fix is to change the underlying Hive client's current database. I think this can fix the issue. 

If we do not reset the current database, the other test cases might still hit the same issue. Thus, this PR tries to do it

Reply:
OK, I see that it changes `currentDb`. Alright, let's try this.